lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -514,7 +514,7 @@ ArrayAttr TypeLoweringVisitor::filterAnnotations(MLIRContext *ctxt,
   if (!annotations || annotations.empty())
     return ArrayAttr::get(ctxt, retval);
   for (auto opAttr : annotations) {
-    std::optional<int64_t> maybeFieldID;
+    std::optional<uint64_t> maybeFieldID;
```

Comment:
Hey, great catch. I'll change this to `getUInt()`. 👍

Reply:
Well, this is slightly more complicated as it means a more thorough audit of how this is stored... I'll see if it's quick otherwise I'll land this and fix in a follow-up.